light-client/src/supervisor.rs

```diff
@@ -25,25 +22,6 @@ pub trait Handle {
     /// Verify to the block at the given height.
     fn verify_to_target(&mut self, height: Height) -> Result<LightBlock, Error>;
 
-    /// Async version of `verify_to_highest`.
-    ///
-    /// The given `callback` will be called asynchronously with the
-    /// verification result.
-    fn verify_to_highest_async(
```

Comment:
Just for context here. The intention of developing an async alternative here was to provide asynchronous verification to the relayer such that it could accumulate events from subscriptions/queries while waiting for verification. We can easily add this back when the requirement materialises in the relayer implementation.

Reply:
Perhaps we could make the API exposed by the Handle a bit more generic by returning the underlying receiver instead of the result of blocking on it. This way, consumers of the API can decide whether they:

a) want the result right away and call `.recv()/.recv_timeout()` on it themselves
b) store the channel away for use at a later time
c) pipe its result into another unbounded channel that will be iterated over at a later time

Reply:
This would allow for both sync and async needs, and give the consumers of the API a bit more control over when they want to access the result of the computation.

Reply:
Moved this discussion in https://github.com/informalsystems/tendermint-rs/pull/402#pullrequestreview-441022995, where I should have posted first thing anyway.